client/state/sites/connection/reducer.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { stubFalse, stubTrue } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { createReducer, keyedReducer } from 'state/utils';
+import {
+	SITE_CONNECTION_STATUS_RECEIVE,
+	SITE_CONNECTION_STATUS_REQUEST,
+	SITE_CONNECTION_STATUS_REQUEST_FAILURE,
+	SITE_CONNECTION_STATUS_REQUEST_SUCCESS,
+} from 'state/action-types';
+
+export const items = createReducer( {}, {
+	[ SITE_CONNECTION_STATUS_RECEIVE ]: ( state, { siteId, status } ) => ( { ...state, [ siteId ]: status } ),
+} );
+
+export const requesting = keyedReducer( 'siteId', createReducer( {}, {
+	[ SITE_CONNECTION_STATUS_REQUEST ]: stubTrue,
+	[ SITE_CONNECTION_STATUS_REQUEST_SUCCESS ]: stubFalse,
+	[ SITE_CONNECTION_STATUS_REQUEST_FAILURE ]: stubFalse,
+} ) );
```

Comment:
Ah, good call. I knew I had to read more about `keyedReducer` before using it 😉 

I've updated this to not use `keyedReducer` so it won't persist - I've also added some tests for that.